[CI] DownsampleRestIT test {p0=downsample/10_basic/Downsample index with multi-value dimensions} failing

**Build Scans:**
- [elasticsearch-intake #36953 / rest-compat](https://gradle-enterprise.elastic.co/s/ewhfmgvbczevk)
- [elasticsearch-pull-request #134347 / rest-compatibility](https://gradle-enterprise.elastic.co/s/ldytpetwoqg3w)
- [elasticsearch-intake #36952 / rest-compat](https://gradle-enterprise.elastic.co/s/psvnx5hoda2uy)
- [elasticsearch-pull-request #134345 / rest-compatibility](https://gradle-enterprise.elastic.co/s/ztfyiqsxco3he)
- [elasticsearch-pull-request #134346 / rest-compatibility](https://gradle-enterprise.elastic.co/s/gyjuf2dryqrvm)
- [elasticsearch-intake #36951 / rest-compat](https://gradle-enterprise.elastic.co/s/xq4ecrrnq2lji)
- [elasticsearch-pull-request #134341 / rest-compatibility](https://gradle-enterprise.elastic.co/s/ir7gxoarq6vxm)
- [elasticsearch-pull-request #134339 / rest-compatibility](https://gradle-enterprise.elastic.co/s/m5beb4hqgmhlo)
- [elasticsearch-pull-request #134335 / rest-compatibility](https://gradle-enterprise.elastic.co/s/yiqbpi6jpivow)
- [elasticsearch-pull-request #134342 / rest-compatibility](https://gradle-enterprise.elastic.co/s/6uamwbhkaihqu)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:downsample:qa:rest:yamlRestCompatTest" --tests "org.elasticsearch.xpack.downsample.DownsampleRestIT.test {p0=downsample/10_basic/Downsample index with multi-value dimensions}" -Dtests.seed=3AC75D75D7B5B198 -Dtests.locale=doi-Deva-IN -Dtests.timezone=Asia/Novosibirsk -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.downsample.DownsampleRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bp0%3Ddownsample%2F10_basic%2FDownsample%20index%20with%20multi-value%20dimensions%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: 
Expected: <21>
     but: was <10>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {p0=downsample/10_basic/Downsample index with multi-value dimensions}
- [main] 3 consecutive failures in step rest-compat
- [main] 10 consecutive failures in step rest-compatibility
- [main] 2 consecutive failures in pipeline elasticsearch-intake
- [main] 3 consecutive failures in pipeline elasticsearch-pull-request
- [main] 13 failures in test test {p0=downsample/10_basic/Downsample index with multi-value dimensions} (13.7% fail rate in 95 executions)
- [main] 3 failures in step rest-compat (100.0% fail rate in 3 executions)
- [main] 10 failures in step rest-compatibility (100.0% fail rate in 10 executions)
- [main] 3 failures in pipeline elasticsearch-intake (11.1% fail rate in 27 executions)
- [main] 9 failures in pipeline elasticsearch-pull-request (15.8% fail rate in 57 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -380,9 +380,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/145573
 - class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeApproximationIT
   issue: https://github.com/elastic/elasticsearch/issues/145461
-- class: org.elasticsearch.xpack.downsample.DownsampleRestIT
-  method: test {p0=downsample/10_basic/Downsample index with multi-value dimensions}
-  issue: https://github.com/elastic/elasticsearch/issues/145578
 - class: org.elasticsearch.xpack.sql.qa.jdbc.single_node.SingleNodeJdbcResultSetIT
   method: testGettingValidLongWithoutCasting
   issue: https://github.com/elastic/elasticsearch/issues/145580

